lib/checks/mobile/css-orientation-lock.js

```diff
@@ -1,134 +1,251 @@
 /* global context */
-
-// extract asset of type `cssom` from context
 const { cssom = undefined } = context || {};
-
-// if there is no cssom <- return incomplete
 if (!cssom || !cssom.length) {
 	return undefined;
 }
 
-// combine all rules from each sheet into one array
-const rulesGroupByDocumentFragment = cssom.reduce(
-	(out, { sheet, root, shadowId }) => {
-		// construct key based on shadowId or top level document
+let isLocked = false;
+let relatedElements = [];
+const rulesGroupByDocumentFragment = groupCssomByDocument(cssom);
+
+for (const key of Object.keys(rulesGroupByDocumentFragment)) {
+	const { root, rules } = rulesGroupByDocumentFragment[key];
+	const orientationRules = getOrientaionCssMediaFeatureRules(rules);
+	if (!orientationRules.length) {
+		continue;
+	}
+
+	orientationRules.forEach(({ cssRules }) => {
+		Array.from(cssRules).forEach(cssRule => {
+			const locked = getIsOrientationLocked(cssRule);
+
+			// if locked and not root HTML, preserve as relatedNodes
+			if (locked && cssRule.selectorText.toUpperCase() !== 'HTML') {
+				const elms =
+					Array.from(root.querySelectorAll(cssRule.selectorText)) || [];
+				relatedElements = relatedElements.concat(elms);
+			}
+
+			// set locked boolean
+			isLocked = locked;
```

Comment:
You are repeatedly setting this. If there are two orientation media queries and the first one has the lock, the second one is going to set this to false again. 

```suggestion
			isLocked = isLocked || locked;
```